src/worker.py

```diff
@@ -171,24 +198,34 @@ async def get_synced_color_depth_outputs(self) -> Tuple[CapturedData, CapturedDa
         )
 
     def get_color_image(self) -> Optional[CapturedData]:
-        color_q = self.color_q_handler.get_queue()
-        try:
-            color_msg = color_q.get(block=True, timeout=5)
-            color_output = self._process_color_frame(color_msg.frame)
-            timestamp = time.time()
-            return CapturedData(color_output, timestamp)
-        except Empty:
-            raise Exception("Timed out waiting for color image data.")
+        color_msg: Optional[FramePacket] = (
+            self.message_synchronizer.get_most_recent_color_msg()
```

Comment:
Big change was to use the message synchronizer to get data as first choice, then fallback to queue if it fails